tt_metal/impl/dispatch/kernels/command_queue_common.hpp

```diff
@@ -53,3 +55,50 @@ tt_l1_ptr const db_cb_config_t* get_remote_db_cb_config(uint32_t base_addr, bool
     const db_cb_config_t* db_cb_config = (db_cb_config_t*)(base_addr + (db_buf_switch * l1_db_cb_addr_offset));
     return db_cb_config;
 }
+
+FORCE_INLINE
+void multicore_cb_push_back(
+    db_cb_config_t* db_cb_config,
+    const db_cb_config_t* remote_db_cb_config,
+    uint64_t consumer_noc_encoding,
+    uint32_t consumer_fifo_16b_limit,
+    uint32_t num_to_write) {
+    db_cb_config->recv += num_to_write;
+    db_cb_config->wr_ptr += db_cb_config->page_size * num_to_write;
+
+    if (db_cb_config->wr_ptr >= consumer_fifo_16b_limit) {
+        db_cb_config->wr_ptr -= db_cb_config->total_size;
+    }
+
+    uint32_t remote_pages_recv_addr = (uint32_t)(&(remote_db_cb_config->recv));
+    noc_semaphore_set_remote((uint32_t)(&(db_cb_config->recv)), consumer_noc_encoding | remote_pages_recv_addr);
+}
+
+FORCE_INLINE
+void multicore_cb_wait_front(db_cb_config_t* db_cb_config, int32_t num_pages) {
+    DEBUG_STATUS('C', 'R', 'B', 'W');
+
+    uint16_t pages_received;
+    do {
+        pages_received = uint16_t(db_cb_config->recv - db_cb_config->ack);
+    } while (pages_received < num_pages);
+    DEBUG_STATUS('C', 'R', 'B', 'D');
+}
+
+void multicore_cb_pop_front(
+    db_cb_config_t* db_cb_config,
+    const db_cb_config_t* remote_db_cb_config,
+    uint64_t producer_noc_encoding,
+    uint32_t consumer_fifo_limit_16B,
+    uint32_t num_to_write,
+    uint32_t page_size_16B) {
+    db_cb_config->ack += num_to_write;
+    db_cb_config->rd_ptr += page_size_16B * num_to_write;
+
+    if (db_cb_config->rd_ptr >= consumer_fifo_limit_16B) {
```

Comment:
need to update all 16B variables to have 16B suffix. Just want to make a note. For example, I'm inferring that since we're comparing consumer_fifo_limit_16B to rd_ptr, rd_ptr should be rd_ptr_16B.